symbolic-symcache/src/writer.rs

```diff
@@ -318,19 +317,22 @@ where
         Ok(())
     }
 
-    /// Adds a function to this SymCache.
+    /// Cleans up a function by recursively removing all empty inlinees, then inserts it into
+    /// the writer. Does nothing if the function is empty itself.
     ///
     /// Functions **must** be added in ascending order using this method. This emits a function
     /// record for this function and for each inlinee recursively.
-    pub fn add_function(&mut self, mut function: Function<'_>) -> Result<(), SymCacheError> {
+    pub fn clean_and_insert_function(
```

Comment:
to be honest, the old name was fine for me.  expanding on the doc comment doesn't do any harm though